### ISSUE TITLE
Expose File as a single field instead of a list for each Version

### ISSUE THREAD

**Issue by diox** [mozilla/addons-server#16465]
Once https://github.com/mozilla/addons-server/issues/16464 is done, `Version` will have a single `File`, all the time. To further clean it up, we'll want to make `File` a `OneToOneField` or even merge `File` and `Version` together... while ensuring that 1) we can deploy without errors (living on the old tag for a short while) 2) the API backwards-compatibility promises are still kept.

**Comment by diox** [comment]
We'll also likely want this at the API level - exposing a single `file` instead of the `files` list for each version. Since we want older versions of the API to remain the same, we'll have to have a compatibility layer no matter what, so that work can be done before and after this issue.

**Comment by ioanarusiczki** [comment]
@diox Can qa do some checks on this issue ?

**Comment by eviljeff** [comment]
QA: this changes the core models (File/Version) so potentially there might be breakage anywhere, but functionally there shouldn't be any noticeable changes.

The API work will be done in #17839 so frontend/API should work and respond as before.
We haven't allowed multiple files for a version in devhub for a while now so there shouldn't be any changes there either, but please double check manage version pages and uploading.
Reviewer tools too - approving and rejecting.

**Comment by ioanarusiczki** [comment]
@eviljeff 
I verified on AMO dev with FF91(Win10):
- detail and version addon APIs - work, I'm getting a 200 status. Also search API (with a query) works as expected.
- I did not notice changes on the addon version page from dev hub in comparison with -stage, I checked on all addon types.
- I uploaded addons and new versions using dev hub and API signing + versions endpoints
- approving and rejecting work (I tested with an admin account)
